packages/js/settings-editor/src/components/settings-group/index.tsx

```diff
@@ -18,24 +25,25 @@ export const SettingsGroup = ( { group }: { group: SettingsGroup } ) => {
 		};
 	};
 	return (
-		<div className="woocommerce-settings-group">
+		<fieldset className="woocommerce-settings-group">
 			<div className="woocommerce-settings-group-title">
 				<Heading level={ 4 }>{ group.title }</Heading>
-				<p dangerouslySetInnerHTML={ sanitizeHTML( group.desc ) } />
+				<legend
+					dangerouslySetInnerHTML={ sanitizeHTML( group.desc ) }
+				/>
 			</div>
 			<div className="woocommerce-settings-group-content">
 				{ group.settings.map( ( setting ) => {
 					const key =
-						setting.id ||
-						setting.type + '-' + setting.title?.replace( '', '-' );
-					return (
-						<div key={ key }>
-							<Heading level={ 5 }>{ setting.title }</Heading>
-							<p>{ setting.type }</p>
-						</div>
-					);
+						setting.id +
+						'-' +
+						( setting.title ?? '' ).replace( ' ', '-' ) +
```

Comment:
I think we should change the `.replace(' ', '-')` to `.replace(/ /g, '-')` to ensure that ALL spaces in data.title are replaced with hyphens, not just the first one. 

```suggestion
						( setting.title ?? '' ).replace( /\s+/g, '-' ) +
```